src/ctapipe/calib/camera/outlier.py

```diff
@@ -0,0 +1,112 @@
+"""
+Outlier detection algorithms to identify faulty pixels
+"""
+
+__all__ = [
+    "OutlierDetector",
+    "RangeBasedOutlierDetector",
+    "MedianBasedOutlierDetector",
+    "StdBasedOutlierDetector",
+]
+
+from abc import abstractmethod
+
+import numpy as np
+
+from ctapipe.core import TelescopeComponent
+from ctapipe.core.traits import List
+
+
+class OutlierDetector(TelescopeComponent):
+    """
+    Base class for outlier detection algorithms.
+    """
+
+    outliers_interval = List(
```

Comment:
In one case, this is the actually clipping interval (in units of the image value), but in others it is a multiplicative factor (unitless). Is it really needed in the base class then? Is there a reason not just just have explicit config variables for each implementation?

Reply:
They follow this approach in `lstchain`, where they have config variables named `charge_median_cut_outliers`, `charge_std_cut_outliers`, `time_cut_outliers`. I aimed here to remove replications and the endless underscore naming conventions (which aren't really precise). The config variable would be the same for the sub-components `MedianBasedOutlierDetector` and `StdBasedOutlierDetector`, so should a parent of those components (maybe named `CameraBasedOutlierDetector`?) be introduced that holds the config variable of the multiplicative factor?

Reply:
This is mainly about properly documenting the options I think. Which is much easier, if you have the option at each class with a fitting help=

Reply:
Yes, It's ok to have it called outliers_interval in each class, but as the meaning is different for each sub-class, it's better to not put it in the base class, but explicitly in each subclass.  And then you can even be more explicit in the names if you prefer, or keep the same name. 

Especiallyy since the configuration for each must be different for each case (i.e. you cannot configure the base class values and then have them work for all sub-classes).  I.e. this would not work:

```yaml
OutlierDetector:
     time_cut_outliers: [3, 5] 
```
And then selecting `outlier_detector_method=MedianOutlierDetector` vs `outlier_detector_method=StdOutlierDetector`, would not work correctly. You'd want to explicitly configure each one.

Reply:
I included the changes. @kosack, can you please have a look if it's now more clear and sounding? Thanks in advance!